qiskit_experiments/curve_analysis/fit_function.py

```diff
@@ -13,11 +13,39 @@
 """
 A library of fit functions.
 """
-# pylint: disable=invalid-name
+# pylint: disable=invalid-name, no-member
+
+import functools
+from typing import Callable, Union
 
 import numpy as np
+from uncertainties import unumpy as unp, UFloat
+
+
+def typecast_returns(fit_func: Callable) -> Callable:
```

Comment:
The name should probably mention `typecast_float`, either that or you could take the type as an arg like:

```python
def typecast_returns(numeric_type: Type) -> Callable:
     # Return the decorator

@typecaset_return(float)
def cos(...)
```